### PR TITLE
tomty black box tests

### PR DESCRIPTION
**Pull request by melezhik**
Example output:

```
$ tomty  --dump-task --color --all
[1/5] / [rak-pager] .......  2 sec. OK
[2/5] / [rak-search] .......  1 sec. OK
[3/5] / [rak-search-before-after] .......  4 sec. OK
[4/5] / [rak-search-dir] .......  2 sec. OK
[5/5] / [rak-search-files-only] .......  2 sec. OK
=========================================
(=: / [5] tests in 11 sec / (5) tests passed
```

### CODE REVIEW

**Comment by melezhik** [comment]
if you need some colors:

output

File: .tomty/rak-search-before-after.raku [hunk 1]
```diff
@@ -0,0 +1 @@
+task-run ".tomty/tasks/rak-search-before-after/";
```
Comment (lizmat) [inline .tomty/rak-search-before-after.raku:1]: Why are these files .raku, while it looks like they're basicall shell ?

Comment (melezhik) [reply .tomty/rak-search-before-after.raku:1]: These are Raku files, it's just in simplest case, they look very alike a bash files, 😁 consider this as more advanced example : 

```raku
task-run ".tomty/tasks/rak-search-before-after/", %(
  lines-before => 10,
  lines-after => 10
);
```

let me improve this bit so you could a real example ...

Comment (melezhik) [reply .tomty/rak-search-before-after.raku]: now it looks like a Raku code )) and it IS a Raku code:

```
$ tomty rak-search-before-after 
10:12:31 07/12/2022 [repository] :: index updated from http://sparrowhub.io/repo/api/v1/index
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: file: 
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before1
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before2
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before3
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before4
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before5
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before6
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before7
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before8
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before9
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_before10
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: foo
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after1
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after2
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after3
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after4
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after5
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after6
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after7
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after8
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after9
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: line_after10
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: <<<
10:12:32 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: execute: rak foo --before=10 --after=10
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: file.txt
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   1:line_before1
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   2:line_before2
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   3:line_before3
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   4:line_before4
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   5:line_before5
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   6:line_before6
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   7:line_before7
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   8:line_before8
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::   9:line_before9
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  10:line_before10
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  11:foo
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  12:line_after1
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  13:line_after2
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  14:line_after3
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  15:line_after4
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  16:line_after5
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  17:line_after6
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  18:line_after7
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  19:line_after8
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  20:line_after9
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] ::  21:line_after10
10:12:33 07/12/2022 [.tomty/tasks/rak-search-before-after/] :: >>>
[task check] stdout match (s) <<<<> True
[task check] stdout match (s) <execute: rak foo --before=10 --after=10> True
[task check] stdout match (s) <file.txt> True
[task check] stdout match (s) <^^\s+ '1:line_before1' \s* $$> True
[task check] stdout match (s) <^^\s+ '2:line_before2' \s* $$> True
[task check] stdout match (s) <^^\s+ '3:line_before3' \s* $$> True
[task check] stdout match (s) <^^\s+ '4:line_before4' \s* $$> True
[task check] stdout match (s) <^^\s+ '5:line_before5' \s* $$> True
[task check] stdout match (s) <^^\s+ '6:line_before6' \s* $$> True
[task check] stdout match (s) <^^\s+ '7:line_before7' \s* $$> True
[task check] stdout match (s) <^^\s+ '8:line_before8' \s* $$> True
[task check] stdout match (s) <^^\s+ '9:line_before9' \s* $$> True
[task check] stdout match (s) <^^\s+ '10:line_before10' \s* $$> True
[task check] stdout match (s) <^^\s*'11:foo' \s* $$> True
[task check] stdout match (s) <^^\s+ '12:line_after1' \s* $$> True
[task check] stdout match (s) <^^\s+ '13:line_after2' \s* $$> True
[task check] stdout match (s) <^^\s+ '14:line_after3' \s* $$> True
[task check] stdout match (s) <^^\s+ '15:line_after4' \s* $$> True
[task check] stdout match (s) <^^\s+ '16:line_after5' \s* $$> True
[task check] stdout match (s) <^^\s+ '17:line_after6' \s* $$> True
[task check] stdout match (s) <^^\s+ '18:line_after7' \s* $$> True
[task check] stdout match (s) <^^\s+ '19:line_after8' \s* $$> True
[task check] stdout match (s) <^^\s+ '20:line_after9' \s* $$> True
[task check] stdout match (s) <^^\s+ '21:line_after10' \s* $$> True
[task check] stdout match (s) <>>>> True
```

**Comment by lizmat** [comment]
FWIW, negating a boolean option can also be done with `--/foo`.  Less chance of typing `False` wrong :-)

**Comment by melezhik** [comment]
| FWIW, negating a boolean option can also be done with --/foo. Less chance of typing False wrong :-)

fixed!

**Comment by melezhik** [comment]
https://gistcdn.githack.com/melezhik/817a5b17b91e1e2fb0aa9ac7f7e4f41c/raw/688f8b1fd1f7e9ff120ad8e2f78bf81e714a360b/report.html

**Pull request closed without merging**